addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/QueuedSend.java

```diff
@@ -0,0 +1,276 @@
+            this.repeatable = repeatable;
+            this.custom_delay_time = custom_delay_time;
+        }
+    }
+
+    BlockingQueue<QueueItem> queue = new LinkedBlockingQueue<>(20);
+    protected final DatagramPacket packet;
+    protected final DatagramSocket datagramSocket;
+    private static final Logger logger = LoggerFactory.getLogger(QueuedSend.class);
+    private int delay_between_commands = 100;
+    private int repeat_commands = 1;
+    private boolean willbeclosed = false;
+    private Thread thread;
+
+    public static final byte NO_CATEGORY = 0;
+
+    /**
+     * Creates a new send queue and starts the background thread.
+     *
+     * @param addr May be null but has to be set by setAddress before first call to queue().
+     * @param port The destination port
+     * @throws SocketException
+     */
+    public QueuedSend(InetAddress addr, int port) throws SocketException {
+        byte[] a = new byte[0];
+        packet = new DatagramPacket(a, a.length, addr, port);
+        datagramSocket = new DatagramSocket();
+        thread = new Thread(this);
+        thread.start();
+    }
+
+    public int getDelayBetweenCommands() {
+        return delay_between_commands;
+    }
+
+    public int getRepeatCommands() {
+        return repeat_commands;
+    }
+
+    public void setRepeatCommands(int repeat_commands) {
+        repeat_commands = Math.max(0, Math.min(5, repeat_commands));
+        this.repeat_commands = repeat_commands;
+    }
+
+    public void setDelayBetweenCommands(int ms) {
+        ms = Math.max(0, Math.min(400, ms));
+        delay_between_commands = ms;
+    }
+
+    /**
+     * The queue process
+     */
+    @Override
+    public void run() {
+        while (!willbeclosed) {
+            QueueItem item;
+            int delay_time = delay_between_commands;
+            try {
+                // block/wait for another item
+                item = queue.take();
+            } catch (InterruptedException e) {
+                if (!willbeclosed) {
+                    logger.error("Queue take failed: " + e.getLocalizedMessage());
+                }
+                break;
+            }
+
+            if (item == null) {
+                continue;
+            }
+
+            if (item.custom_delay_time != 0) {
+                delay_time = item.custom_delay_time;
+            }
+
+            packet.setData(item.data);
+            try {
+                int repeat_remaining = item.repeatable ? repeat_commands : 1;
+                repeat_remaining = Math.min(repeat_remaining, 1);
+
+                StringBuffer s = new StringBuffer();
+                for (int i = 0; i < item.data.length; ++i) {
+                    s.append(String.format("%02X ", item.data[i]));
+                }
+
+                while (repeat_remaining > 0) {
+                    datagramSocket.send(packet);
+                    --repeat_remaining;
+                    // logger.debug("Sent packet '{}' to bridge {}", s.toString(),
```

Comment:
Is there a reason the debug logging is commented out?

Using RC5 my lights aren't switching as reliably as using my 2.0.0 binding modified to  repeat 4 times. It would be good to be able to verify the repeats/delay is working as expected.